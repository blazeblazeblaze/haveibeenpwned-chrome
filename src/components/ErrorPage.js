import React from 'react';
import PropTypes from 'prop-types';
import RestartLink from './RestartLink';

const ErrorPage = ({errors, onStartOver}) => {

  return(
    <div>
      <span className="App-intro">
        <p>
          Oh no! It looks like you've encountered errors.
          {errors.map((error)=> <span key={error}>{error}</span>)}
        </p>
        <RestartLink onStartOver={onStartOver} />
      </span>
    </div>
  )
}

ErrorPage.propTypes = {
  errors: PropTypes.array.isRequired,
  onStartOver: PropTypes.func.isRequired
};

ErrorPage.defaultProps = {
  errors: []
};

export default ErrorPage;
