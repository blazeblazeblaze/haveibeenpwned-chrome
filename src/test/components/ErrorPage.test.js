import React from 'react';
import ReactDOM from 'react-dom';
import ErrorPage from '../../components/ErrorPage';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ErrorPage onStartOver={jest.fn} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('ErrorPage renders with errors', () => {
  const errors = ['API timeout'];
  const component = renderer.create(
    <ErrorPage onStartOver={jest.fn} errors={errors} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
