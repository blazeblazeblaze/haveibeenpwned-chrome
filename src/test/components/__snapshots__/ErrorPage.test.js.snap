// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage renders with errors 1`] = `
<div>
  <span
    className="App-intro"
  >
    <p>
      Oh no! It looks like you've encountered errors.
      <span>
        API timeout
      </span>
    </p>
    <a
      className="pure-button pure-button-primary button-small"
      href=""
      onClick={[Function]}
    >
      Try with a different email
    </a>
  </span>
</div>
`;
